managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2957,7 +2998,12 @@ public void asyncOffloadPrefix(Position pos, OffloadCallback callback, Object ct
             promise.whenComplete((result, exception) -> {
                 offloadMutex.unlock();
                 if (exception != null) {
-                    callback.offloadFailed(new ManagedLedgerException(exception), ctx);
+                    Throwable t = FutureUtil.unwrapCompletionException(exception);
```

Comment:
why don't we call `asyncRefreshLedgersInfoOnBadVersion` here but in `offloadPrefix`? 
`asyncOffloadPrefix` is a public method.

Reply:
Great catch! We need to do the refresh here